### PR TITLE
Jm overview stretch1

### PR DESCRIPTION
**Pull request by Greenfin17**
# Overview with some stretch items

## Description
Displays sidebar bio, pinned and un-pinned repose, and minor enhanced bio at the page top.

## Related Issue
https://github.com/nss-evening-cohort-14/gitsub-team-no-name/issues/26

## Motivation and Context
This contains MVP features required as well as some stretch goals.

## How Can This Be Tested?
Pull down the code and test that you can pin and un-pinned repositories, as well as add repositories with technologies.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -217,7 +224,7 @@ const bioBadges = (userBio) => {
     `<h4>Highlights</h4>
      <ul class="highlights">`;
   for(let i = 0; i< userBio.badges.length; i++) {
-    console.log(badges[userBio.badges[i]].badgeName);
+    //console.log(badges[userBio.badges[i]].badgeName);
```
Comment (BenJarrett) [inline main.js:227]: remove this console log <3

File: main.js [hunk 2]
```diff
@@ -228,7 +235,7 @@ const bioOrganizations = (userBio) => {
     `<h4>Organizations</h4>
      <ul class="organizations">`;
   for(let i = 0; i < userBio.organizations.length; i++) {
-    console.log(organizations_arr[userBio.organizations[i]].orgName);
+    //console.log(organizations_arr[userBio.organizations[i]].orgName);
```
Comment (BenJarrett) [inline main.js:238]: remove console log <3

File: main.js [hunk 3]
```diff
@@ -250,11 +257,72 @@ const paintBio = (userBio) => {
...
+      <p class="card-text about-text">${developer.longBioText}</p>
+    </div>`;
+  printToDom("#about-developer", aboutString);
+}
+
+
+
+//modal dialog box from bootstrap components
+const pinnedDialog = (developer) => {
+  //console.log(developer.keyId);
```
Comment (BenJarrett) [inline main.js:281]: remove console log <3

File: main.js [hunk 4]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+  repoString += `<div class="repo-dialog" id="pinned-dialog"></div>`;
+  repoString += pinnedDialog(developer);
+  let color = '';
   let techIndex = 0; //tells us the most prominent tech used in the repository
   let i = 0; //limit to six repositories
   for(let item of developer.repositories) {
-    if(i < 6) {
+    if(i < 6 && ((item.pinned && has_pinned) || !has_pinned)) {
       techIndex = item.technologies[0];
+      //console.log("tech index: " + techIndex);
```
Comment (BenJarrett) [inline main.js:362]: remove console log <3

File: main.js [hunk 5]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+      color = technologies_arr[techIndex].color;
+      //console.log(color);
```
Comment (BenJarrett) [inline main.js:364]: remove console log <3

**Review by elliewalkersm — APPROVED** [review]
Looks good to me!

File: main.js [hunk 6]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
       //console.log(technologies[techIndex].name);
```
Comment (BenJarrett) [inline main.js:365]: remove console log <3

File: main.js [hunk 7]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
   }
   repoString += `</div>`;
   printToDom('#gitHubRepos', repoString);
 }
 
+//updates which repositories are pinned using the checkbox form.
+const savePinned = (e) => {
+  //get id of form.
+  let develId = e.srcElement.parentElement.parentElement.id;
+  //console.log("develId = " + develId);
```
Comment (BenJarrett) [inline main.js:388]: remove console log <3

File: main.js [hunk 8]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+  let pinned_form = document.getElementById(develId);
+  //extract developer keyId from id of form.
+  let develKey = (develId.substr(6));
+  let element = '';
+  for(let item of developers_arr[develKey].repositories){
+    //console.log(item.repoID);
```
Comment (BenJarrett) [inline main.js:394]: remove console log <3

File: main.js [hunk 9]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+    //get the corresponding checkbox for each repository
+    element = document.getElementById(`checkbox_${item.repoID}`);
+    //console.log(element);
```
Comment (BenJarrett) [inline main.js:397]: remove console log <3

File: main.js [hunk 10]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+    if(element.checked) {
+      item.pinned = true;
+    } else {
+      item.pinned = false;
+    }
+    //console.log(item);
```
Comment (BenJarrett) [inline main.js:403]: remove console log <3

File: main.js [hunk 11]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+    case "save-pinned-dialog" :
+      savePinned(e);
+      document.querySelector('#pinned-modal').style.display = "none";
+      break;
+  }
+}
+
+//fetch the technologies saved
+const saveTechnologiesSelections = ()  => {
+//  console.log(e.srcElement.parentElement.parentElement.id);
```
Comment (BenJarrett) [inline main.js:427]: remove console log <3

File: main.js [hunk 12]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+  let checkbox = ''; 
+  for(item of technologies_arr) {
+    checkbox = document.getElementById(`tech-checkbox_${item.techID}`);
+    //console.log(item.techID);
+    if(checkbox.checked) {
+      tempTechnologies.push(item.techID);
+    }
+  }
+  //console.log(tempTechnologies);
```
Comment (BenJarrett) [inline main.js:436]: remove console log <3

File: main.js [hunk 13]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+const clearNewRepositoryForm = () => {
+  clearCheckBoxes();
+  document.getElementById("project-name").value = '';
+  document.getElementById("project-description").value = '';
+}
+
+
+//Event listener for new repository form
+const listenNewRepository = (e) => {
+  //console.log(e.target.id);
```
Comment (BenJarrett) [inline main.js:457]: remove console log <3

File: main.js [hunk 14]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+      } else if(list.style.display === "block") {
+        list.style.display = '';
+      } 
+      break;
+    case "x-close-tech-dialog" :
+    case "close-tech-dialog"   :
+      list.style.display = '';
+      break;
+    case "save-tech-dialog" :
+      //console.log("save-tech-dialog hit");
```
Comment (BenJarrett) [inline main.js:473]: remove console log <3

File: main.js [hunk 15]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+//Add pinned repository
+const repositorySubmit = (e) => {
+  if(e.target.id == 'repository-submit') {
+    let repoName = document.querySelector('#project-name').value;
+    let repoDescription = document.querySelector('#project-description').value;
+    if(repoName && repoDescription) {
+      let pinned = true;
+      let technologies = [0]; //default technology
+      if(tempTechnologies.length) {
+        //console.log("has technologies");
```
Comment (BenJarrett) [inline main.js:560]: remove console log <3

File: main.js [hunk 16]
```diff
@@ -270,67 +338,260 @@ const hasPinned = (developer) => {
...
+        //console.log(tempTechnologies);
+        technologies = [...tempTechnologies]; //save technologies from modal dial
+      } 
+      tempTechnologies.length = 0; //Clear the array
+      let repoID = developers_arr[0].repositories.length; 
+      //console.log(repoID);
```
Comment (BenJarrett) [inline main.js:566]: remove console log <3

**Review by BenJarrett — APPROVED** [review]
Great Job! This is incredible!

**Review by Mitchell-Crumbley — APPROVED** [review]
J'approve

**Pull request merged**